presto-main/src/main/java/com/facebook/presto/server/protocol/Query.java

```diff
@@ -248,7 +248,7 @@ public synchronized boolean isClearTransactionId()
         return clearTransactionId;
     }
 
-    public synchronized ListenableFuture<QueryResults> waitForResults(OptionalLong token, UriInfo uriInfo, Duration wait)
+    public synchronized ListenableFuture<QueryResults> waitForResults(OptionalLong token, UriInfo uriInfo, String scheme, Duration wait)
```

Comment:
I dislike passing scheme separately since we already have `UriInfo` and it should logically be part of that. Unfortunately, there is no easy way to create a modified `UriInfo`. We could instead change this to take a `Supplier<UriBuilder>` that does the replacement.

However, this is a bigger change, so I'm fine not doing it here.

Reply:
Agreed, that would be nicer.